share/functions/__fish_print_hostnames.fish

```diff
@@ -62,10 +62,11 @@ function __fish_print_hostnames -d "Print a list of known hostnames"
                 | string trim | string replace -r -a '\s+' ' ')
             end
 
+            cd $relative_path
```

Comment:
I'm always nervous with changing the current directory in completions & functions; it mucks up the directory stack, for one thing, and there's definitely been problems with people overloading the cd command. No, they shouldn't do that, but we shouldn't leave landmines like this behind.